internal/metrics/collectors/controller.go

```diff
@@ -21,12 +21,13 @@ type ControllerMetricsCollector struct {
 }
 
 // NewControllerMetricsCollector creates a new ControllerMetricsCollector
-func NewControllerMetricsCollector(crdsEnabled bool) *ControllerMetricsCollector {
+func NewControllerMetricsCollector(crdsEnabled bool, ingressClassLabel map[string]string) *ControllerMetricsCollector {
```

Comment:
similar to the comment in `main.go`, it is not necessary to use `map` to represent just one label value. Either change the type to `string` or change to name to `constLabels`. With the former, we would be able to easily add more const labels in the future. 

Same applies to `manager.go ` and `listener.go`.